src/isar/state_machine/states/finalize.py

```diff
@@ -36,3 +39,11 @@ def _log_state_transitions(self):
         )
 
         self.logger.info(f"State transitions:\n  {state_transitions}")
+
+    def _update_tasks(self):
+        for task in self.state_machine.current_mission.tasks:
+            for step in task.steps:
+                if step.status == StepStatus.NotStarted:
+                    step.status = StepStatus.Cancelled
+            if task.status == StepStatus.NotStarted:
+                task.status = StepStatus.Cancelled
```

Comment:
Should we already now introduce seperate enums for `StepStatus` and `TaskStatus`?